### ISSUE TITLE
System.IO.IOException: 'Unable to read data from the transport connection: 

### ISSUE THREAD

**Issue by uno-studio** [Fusion86/ScrcpyNet#3]
System.IO.IOException: 'Unable to read data from the transport connection: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond..'

![Image 1](https://user-images.githubusercontent.com/43975116/146732682-4de746f2-ced4-4607-a8a4-19cb9f65c7d4.png)

**Comment by Fusion86** [comment]
This might be expected behavior, because when starting scrcpy it takes a few moments to establish the connection. The `catch (IOExpection ex)` line should ignore that exception, but for some reason your debugger still catches it. Could you try this again with the default Visual Studio debugger exception settings, or with no debugger at all.

![devenv_2022_01_04_21_19_574](https://user-images.githubusercontent.com/4460428/148118908-65acf02c-2cf2-4e1c-b103-cb547e9421bb.png)

**Comment by unofficialdev** [comment]
> This might be expected behavior, because when starting scrcpy it takes a few moments to establish the connection. The `catch (IOExpection ex)` line should ignore that exception, but for some reason your debugger still catches it. Could you try this again with the default Visual Studio debugger exception settings, or with no debugger at all.
> 
> ![devenv_2022_01_04_21_19_574](https://user-images.githubusercontent.com/4460428/148118908-65acf02c-2cf2-4e1c-b103-cb547e9421bb.png)

glad you have released nuget packages, I still haven't tried scrcpynet again after many failed attempts. i wonder if you should refer to qtscrcpy? qtscrcpy has rewritten the scrcpy client based on c++, the application works great and is open source.

**Comment by unofficialdev** [comment]
even i just retried it with wpf sample app. the application exits as soon as the first frame is received
![Image 1](https://user-images.githubusercontent.com/43975116/149653973-0e3af199-9750-4818-9862-bf4bc68411a9.png)

**Comment by Fusion86** [comment]
@unofficialdev I saw that you replied to this issue, but then deleted your comments? Are you still having issues?

**Comment by uno-studio** [comment]
after retrying many times i succeeded, but still can't catch that problem, i think you fixed it, so ignore

**Comment by uno-studio** [comment]
> @unofficialdev Tôi thấy rằng bạn đã trả lời về vấn đề này, nhưng sau đó đã xóa nhận xét của bạn? Bạn vẫn gặp sự cố?

after a few days of stable operation, today I got the application exit error "exception code: 0xc0000374" again.
`The program '[16396] ScrcpyNet.Sample.Wpf.exe' has exited with code 3221226356 (0xc0000374).`
This error occurs after receiving the first image frame. no error message, don't know where the error occurred. but i got this error code so many times. google search brought me to a keyword "STATUS_HEAP_CORRUPTION" (Heap corruption is typically caused by unmanaged code). Can you investigate this further? i guess there was a bug somewhere in the VideoStreamDecoder or a bug with the Ffmpeg.AutoGen library. Because after remove the video thread, this error doesn't appear

**Comment by uno-studio** [comment]
@Fusion86 please check

**Comment by Fusion86** [comment]
Yeah I saw your reply, unfortunately I currently don't have the time to work on this.
